viewing_party/party.py

```diff
@@ -1,23 +1,201 @@
+                user_unique_movies.append(movie)
+
+    return user_unique_movies
+
+def get_friends_unique_watched(user_data):
+    
+# Consider the movies that the user has watched, 
+# and consider the movies that their friends have watched. 
+# Determine which movies at least one of the user's friends have watched, 
+# but the user has not watched.
+# Return a list of dictionaries, that represents a list of movies
+    
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+
+    friends_unique_movies = []
+
+    friends_movie_watched = []
+
+    # Big O notation of O(n^2), try to refactor
+    for friend in friends:
+        for i in range(len(friend["watched"])):
+            friends_movie_watched.append(friend["watched"][i]) 
+
+    for movie in friends_movie_watched:
+        if movie not in user_watched:
+            if movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+# Determine a list of recommended movies. 
+# A movie should be added to this list if and only if:
+# - The user has not watched it
+# - At least one of the user's friends has watched
+# - The `"host"` of the movie is a service that is in 
+# - the user's `"subscriptions"`
+# Return the list of recommended movies
+
+    recommendations = []
+    
+    for movie in get_friends_unique_watched(user_data):
+        if movie["host"] in user_data["subscriptions"]:
+            recommendations.append(movie)
+
+    return recommendations
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    
+# Consider the user's most frequently watched genre. 
+# Then, determine a list of recommended movies. 
+# A movie should be added to this list if and only if:
+# - The user has not watched it
+# - At least one of the user's friends has watched
+# - The `"genre"` of the movie is the same as the user's most frequent genre
+# Return the list of recommended movies
+
+    recommendations = []
+
+    for movie in get_friends_unique_watched(user_data):
+        if movie["genre"] == get_most_watched_genre(user_data):
+            recommendations.append(movie)
+
+    return recommendations
+
+def get_rec_from_favorites(user_data):
+
+# Determine a list of recommended movies. 
+# A movie should be added to this list if and only if:
+# - The movie is in the user's `"favorites"`
+# - None of the user's friends have watched it
+# Return the list of recommended movies
+    
+    recommendations = []
+
+    for movie in user_data["favorites"]:
+        if movie in get_unique_watched(user_data):
+            recommendations.append(movie)
+
+    return recommendations
```

Comment:
Great Job! You both did an amazing job with this project! Please make sure to reach out if you have any questions about the comments I left. Also, if you are wondering about why I didn't leave many comments about `Time Complexity` is that due to the nature of this project and the placement of it I felt as if it would be be unfair since this is you all's first time working with nested loops and data structures (Though I did leave a few that may have been high priority).  ⭐️